lib/appsignal/transaction.rb

```diff
@@ -126,11 +127,86 @@ def set_tags(given_tags = {})
       @tags.merge!(given_tags)
     end
 
+    # Set a custom action name for this transaction.
+    #
+    # When using an integration such as the Rails or Sinatra AppSignal will try
+    # to find the action name from the controller or endpoint for you.
+    #
+    # If you want to customize the action name as it appears on AppSignal.com
+    # you can use this method. This overrides the action name AppSignal
+    # generates in an integration.
+    #
+    # @example in a Rails controller
+    #   class SomeController < ApplicationController
+    #     before_action :set_appsignal_action
+    #
+    #     def set_appsignal_action
+    #       Appsignal.set_action("DynamicController#dynamic_method")
+    #     end
+    #   end
+    #
+    # @param action [String] the action name to set.
+    # @return [void]
     def set_action(action)
       return unless action
+      @action = action
       @ext.set_action(action)
     end
 
+    # Set an action name only if there is no current action set.
+    #
+    # Commonly used by AppSignal integrations so that they don't override
+    # custom action names.
+    #
+    # @example
+    #   Appsignal.set_action("foo")
+    #   Appsignal.set_action_if_nil("bar")
+    #   # Transaction action will be "foo"
+    #
+    # @param action [String]
+    # @return [void]
+    # @see #set_action
+    def set_action_if_nil(action)
+      return if @action
+      set_action(action)
+    end
+
+    # Set a custom namespace for this transaction.
+    #
+    # When using an integration such as Rails or Sidekiq AppSignal will try to
+    # find a appropriate namespace for the transaction.
+    #
+    # A Rails controller will be automatically put in the "http_request"
+    # namespace, while a Sidekiq background job is put in the "background_job"
+    # namespace.
+    #
+    # Note: The "http_request" namespace gets transformed on AppSignal.com to
+    # "Web" and "background_job" gets transformed to "Background".
+    #
+    # If you want to customize the namespace in which transactions appear you
+    # can use this method. This overrides the namespace AppSignal uses by
+    # default.
+    #
+    # A common request we've seen is to split the administration panel from the
```

Comment:
I'd change panel to "administration routes" or "administration namespace"? Pannel suggests it's a part of the main app, while the purpose of namespaces is to separate... namespaces :)